integration_tests/__tests__/snapshot-test.js

```diff
@@ -41,4 +49,74 @@ describe('Snapshot', () => {
       output['snapshot is not influenced by previous counter 0']
     ).not.toBe(undefined);
   });
+
+  describe('validates existance of snapshots correctly', () => {
+    const pathToOriginal = path.resolve(
+      __dirname,
+      '../snapshot/__tests__/snapshot.js'
+    );
+    const originalContent = String(fs.readFileSync(pathToOriginal));
+    const pathToCopy = pathToOriginal.replace('.js', '_copy.js');
+
+    it('delete the snapshot if the test file has been removed', () => {
+      fs.writeFileSync(pathToCopy, originalContent);
+      const firstRun = runJest.json('snapshot', []);
+      fs.unlinkSync(pathToCopy);
+
+      const secondRun = runJest.json('snapshot', []);
+
+      expect(firstRun.json.numTotalTests).toBe(6);
+      expect(secondRun.json.numTotalTests).toBe(3);
+      expect(fileExists(snapshotCopy)).toBe(false);
+    });
+
+    it('delete the snapshot when a test does removes all the snapshots', () => {
+      fs.writeFileSync(pathToCopy, originalContent);
+      const firstRun = runJest.json('snapshot', []);
+      fs.unlinkSync(pathToCopy);
+
+      fs.writeFileSync(pathToCopy, '');
```

Comment:
we are changing Jest so that it throws when a test file contains no tests, so it would be nice to change this so we won't have to update this test later.